docs/design/task-launch-order-within-job.md

```diff
@@ -0,0 +1,98 @@
+| ---------- | -------- | ------------ | -------- | -------------------------------------------------- | -------------------------- |
+| simple ETL | 3        | linear dependencies | no     | 1. One VCJob per phase<br />2. Manage vcjob with argo | 1.Using external solutions<br />2.Uniform management of resources across the operation is not possible<br />3.Cannot handle Running state dependencies |
+| complex ETL | n>=3     | figure dependency | Serial-parallel combination | 1. One VCJob per phase<br />2. Manage vcjob with argo| 1.Using external solutions<br />2.Uniform management of resources across the operation is not possible<br />3.Cannot handle Running state dependencies |
+| MPI        | 2        | linear dependencies | yes    | 1. Add initcontainer to the task<br/>2.Using user-written check scripts in initcontainer | 1.Increase the cost of use for users<br />2.Still using resources while waiting |
+| pytorch    | 2        | linear dependencies | yes    | 1. Add initcontainer to the task<br/>2.Using user-written check scripts in initcontainer |1.Increase the cost of use for users<br />2.Still using resources while waiting|
+| matlab     | 2        | linear dependencies | yes    | 1. Add initcontainer to the task<br/>2.Using user-written check scripts in initcontainer |1.Increase the cost of use for users<br />2.Still using resources while waiting|
+| tensorflow | n>=2   | linear dependencies | yes    | 1. Add initcontainer to the task<br/>2.Using user-written check scripts in initcontainer |1.Increase the cost of use for users<br />2.Still using resources while waiting|
+
+## Requirement
+
+Based on the scenarios listed above, the dependencies can be abstracted as:
+
+* task-A depends on task-B, which means A must be started first and then B.
+* Triggering policy, for example level triggering and edge triggering when handling circuit. In our cases, the triggering policy may be the task's event, like task running/restarted/completed. 
+
+For the ease of end user's experince, we need to unify the way of composing a complicated jobs with lots of tasks, instead of letting user handle the complexcity themselvs using init-containers or other workflow tools. So we need to  have an more advanced VCJob that has below abilities:
+
+* Each task can have one or more tasks it depends on, and it can specify the triggering policy.
+* Examples of triggering policys can be:
+  * task starts to run
+  * task restarts
+  * task completes
+* User may specify the time period after the triggering event happens, for example
+  * Imediately after task completed
+  * Half an hour afater task starts to run or completes
+* Besides the time period, user may specify an certain condition as the triggering policy, for example,
+  * Check if a specified file exists in the /data directory.
+  * Check if the specified port is open, such as 2222.
+  * Check if the HTTP GET request to the specified path was successfully accessed, such as /health.
+  
+## Design
```

Comment:
1.  Add the feature interaction with volcano existing feature.
2.  Is there any impact to existing feature? please add these info to the design doc.
3.  Pleaes add the desgin behavior (usage) for user in doc.